### ISSUE TITLE
Param description as optional mode

### ISSUE THREAD

**Issue by Sulpher** [joomla/joomla-cms#35580]
In Joomla 4 params have no description.
As I've heard, it happened to meet WCAG standards. 

However, param descriptions are a must-have to understand what exact param is used for especially when it comes with new params that J3 doesn't have or when a user is a newbie. And, to tell the truth, sometimes even experienced users need to see the description to get more info about technical details.

My idea is to add [QUESTION] icon near each param and make it clickable. When the click is performed, the tooltip window appears.
And this mode can be optional - so if someone needs it - they can disable tooltips and get the result as now.

Joomla 4:
<img width="1352" alt="Снимок экрана 2021-09-17 в 14 24 13" src="https://user-images.githubusercontent.com/3432048/133775531-2f47ee11-b809-44de-9258-86961e947171.png">

Joomla 3:
![Снимок экрана 2021-09-17 в 14 24 14](https://user-images.githubusercontent.com/3432048/133775547-898d0f90-36b1-4a5a-ae61-55c194f41bca.jpeg)

**Comment by brianteeman** [comment]
1. Tooltips were NOT removed because of accessibility. They could have been made accessible
2. In Joomla 3 there was no indicator if something had a tooltip so everything did. 90% of which were completely useless. Making the job of translaters much much harder
3. In Joomla 4 we decided to show "tooltips" permanently where needed. As can be seen here
![image](https://user-images.githubusercontent.com/1296369/133777679-93bc496e-33cd-4dfa-a6bb-20f6f4615fd2.png)
4. "Tooltips" should provide additional information to provide clarity on the purpose of the field. They should not replace a help document

Did we get it 100% right. Probably not but that is why we had beta versions and release candidates so that people could test and give feedback. If my memory serves me correctly it is three years since the change was made.

**Comment by PhocaCz** [comment]
> 90% of which were completely useless

I don't mean to be annoying, just think of it as a technical question. Was there any research done on this, a poll, or did someone simply say it would be better not to display certain information?

For me personally, it's very inconsistent and confusing because I don't see any key in when to display the hint or not. What I see as systematic and not confusing is that the tooltip will also display information that most people don't need to know.

In fact, the tooltip has one great feature. Those who want to see it, will go to the tooltip. Those who don't want to see it won't see anything. People just know, where to find the quick help.

> Probably not but that is why we had beta versions and release candidates so that people could test and give feedback.

Does it mean, it is not allowed to give feedback for stable version?

I will give myself as an example. I am only now getting feedback from community members. Either I can tell these members that it is too late, or I am simply trying to provide this feedback even now. It's not always about the person giving the feedback, but also about the person collecting more feedback in their community, and it comes gradually and later.

> If my memory serves me correctly it is three years since the change was made.

Another very important thing is that the extensions in which this issue shows up in full are only available from the stable release. I would strongly urge everyone to not only look at features from the **core** perspective, but also from the **extension** perspective. Something that's not so prominent in the core can be very problematic in extensions.

I don't want to cause conflicts, please don't take this personally, but if I played a game with my users like "90% is useless, you should have called when Beta was, etc." I wouldn't have any more users :-( Why I'm talking about this. Because I am part of various Joomla communities where people are leaving Joomla as a system because nobody listens to them.

BTW - I am not writing this comment as myself, rather I am speaking for the people in the communities in which I am a member. For me personally, such a shortage is more of an opportunity to create another extension that people will thank me for.

**Comment by brianteeman** [comment]
> ecause I don't see any key in when to display the hint or not. What I see as systematic and not confusing is that the tooltip will also display information that most people don't need to know.

Display it when its useful. Dont when its not.

Just looking at your own downloads extension you can see how many of the field descriptions are completely useless and of no help to anyone at all. So you really think it is helpful to make your users hover over every field to see if there is some hidden magic only to read such useless text. They will quickly realise they are useless and stop even looking - and then you as a developer will wonder why people dont understand your extension

Just picking one field as an example
![image](https://user-images.githubusercontent.com/1296369/133783334-e175b9e2-2907-48f1-a9ec-855d222e619b.png)

> I don't mean to be annoying, just think of it as a technical question. Was there any research done on this, a poll, or did someone simply say it would be better not to display certain information?

Yes I said it was useless to display useless information. It was merged very very early so that you could give feedback on it. It was not something changed at the last minute. The beta and rc releases are the research.

> Does it mean, it is not allowed to give feedback for stable version?

No one stops you but then again your opinion would have been more useful earlier

> Something that's not so prominent in the core can be very problematic in extensions.

Not at all.  None of your descriptions are hidden now. They are all displayed for your users to see and they dont have to hover over field one at a time to see them. 

YES I AM ANGRY WHEN CODE COMMITTED THREE YEARS AGO IS COMPLAINED ABOUT AS IF IT WAS COMMITTED YESTERDAY.

**Comment by PhocaCz** [comment]
> YES I AM ANGRY WHEN CODE COMMITTED THREE YEARS AGO IS COMPLAINED ABOUT AS IF IT WAS COMMITTED YESTERDAY.

Did you read what I wrote? It's a reality that you don't get feedbacks from other members until the stable version. Yes, the reality is that 99% of the members in all the communities where I am a member only started testing the version when it was stable. And someone is simply listening to those members and conveying that feedback only now. And either we want to hear it or we ignore it. But the fact is that simply based on various events, this feedback is only coming now. That's just the reality.

It's just feedback, an opinion. It's not something to upset anyone.

**Comment by Sulpher** [comment]
> Did we get it 100% right. Probably not but that is why we had beta versions and release candidates so that people could test and give feedback. If my memory serves me correctly it is three years since the change was made.

Well, yes - some param descriptions can be shorter or do not need to be displayed. However, there are some params that really need a bigger description. E.g. params in Global configuration. All params matter!
The users won't go to the documentation if something is not yet clear. They will skip such params and the level of knowledge be reduced for the average new user. 


<img width="576" alt="Снимок экрана 2021-09-17 в 16 08 27" src="https://user-images.githubusercontent.com/3432048/133787916-669ad6ee-e286-44f7-bc3b-7b93f482f00f.png">

Sorry but I think it is always never too late to report bugs and suggest improvements. Just remember how Joomla 3 was changed from the beginning and how many cool featured were implemented later.
The persons representing the project should endorse everyone who spends their time testing the CMS and writing the feedback, sooner or later. Joomla is a free platform and everyone work on it when they have time/motivation/etc. 
I think it is good when the project gets feedback. It shows that the project is live and active.

**Comment by dgrammatiko** [comment]
> I think it is good when the project gets feedback.

The change WAS DONE 3 YEARS AGO where was your feedback all these years?

Also, since I was involved in this change, I kept repeating this quote: `A user interface is like a joke. If you have to explain it, it's not that good.` - Martin LeBlanc

Also another quote from Elon Musk: 
`Some people don't like change, but you need to embrace change if the alternative is disaster.`

**Comment by PhocaCz** [comment]
> The change WAS DONE 3 YEARS AGO where was your feedback all these years?

Please read:
> Did you read what I wrote? It's a reality that you don't get feedbacks from other members until the stable version. Yes, the reality is that 99% of the members in all the communities where I am a member only started testing the version when it was stable. And someone is simply listening to those members and conveying that feedback only now. And either we want to hear it or we ignore it. But the fact is that simply based on various events, this feedback is only coming now. That's just the reality.

> Another very important thing is that the extensions in which this issue shows up in full are only available from the stable release. I would strongly urge everyone to not only look at features from the core perspective, but also from the extension perspective. Something that's not so prominent in the core can be very problematic in extensions.

> I will give myself as an example. I am only now getting feedback from community members. Either I can tell these members that it is too late, or I am simply trying to provide this feedback even now. It's not always about the person giving the feedback, but also about the person collecting more feedback in their community, and it comes gradually and later.

All written above.

You really don't understand that many many people started to test Joomla 4 now?

What exactly don't you understand about the user feedback only coming in now for various reasons?

How do I understand that instead of arguing about the use of a property, we deal with what time the feedback came?

Is there a reason we're diverting the conversation away from factual comments to personal attacks? Why should someone who provides feedback has to answer personal questions about why he/she didn't provide that feedback three years ago?

**Comment by Sulpher** [comment]
> > I think it is good when the project gets feedback.
> 
> The change WAS DONE 3 YEARS AGO where was your feedback all these years?

Damn, you're absolutely unbearable!
Joomla does not pay money and cannot demand users to be volunteers. That's why it took 7 years for J4 to be released.
I am obliged neither you nor the community to take part WHEN YOU THINK it's good.  I do it in my spare time when I have a mood. I hope it is directly clear.
The community members SHOULD BE thankful to users for the feedback and the time spent on the project. 

Leave your imagined world with pink ponies and unicors  (how it should be) and return to reality (how it is).

**Comment by dgrammatiko** [comment]
> You really don't understand that many many people started to test Joomla 4 now?

I do understand that

> Another very important thing is that the extensions in which this issue shows up in full are only available from the stable release.

Let me ask you something: did you put the energy required to bring your extension to the J4 level? To me this complaint is all about `my old forms were fine and now they look bad`. But it's not Joomla's fault, the project decided to simplify by removing cluttered information AND make prominent all the needed information. Of course, you have to ADJUST your forms to comply with the new format by better naming your labels and use efficient descriptions wherever these are needed.
The part that you need information for every field goes from UI to Documentation and if you have to roll the documentation embedded in the form, then something is really wrong and that definitely is not the platform...

**Comment by PhocaCz** [comment]
It's good to know that some extensions don't include simple parameters like whether or not to publish an article. But also parameters for which there are many pages of documentation (just for the parameter itself), and the short version used in the tooltip contains dozens/hundreds of words because it is in context with other parameters. This has nothing to do with personal stuff like "my old form". If you knew my extensions, you would know that I try to adapt them to the latest features of the system, unlike many other developers.

This is a general suggestion about simply being able to test some things when the system was stable. Because it was only then that various extensions were available that contained different features than core.

But it depends on everyone, my advice is that when I create a feature, I don't just look at how it will work in core, I also look at how it will work in extensions or in other parts. I'm not just thinking about making sure that parameters are easy to explain, but also that some parameters, even if they have extensive documentation, unfortunately have to have extensive information in the tooltip.

Whether the person who does such propriety wants to hear that is up to them. However, it would be good to make a factual arguments.

But let's leave the extension aside, in the first question I see the core parameter "Enable versions" in the images as an example. I really don't think I can find any Joomla user who would appreciate the fact that now he/she has to visit external documentation for a parameter that says nothing.

In abbreviated computer speech:

Tooltip:
a) I need short explanation for this parameter - great, it is here
b) I don't need short explanation for this parameter - great, it is not here, because I didn't hover the parameter title

No description at all:
a) I need short explanation for this parameter - hmmm, nothing there, I need to search web
b) I don't need short explanation for this parameter - great, it is not there.

And again, this is an opinion, an opinion at the time we came across this problem. If it's too late or you don't like this opinion, maybe it's better to leave it without comment.

**Comment by kernusr** [comment]
> 
> 
> > I think it is good when the project gets feedback.
> 
> The change WAS DONE 3 YEARS AGO where was your feedback all these years?
> 
> Also, since I was involved in this change, I kept repeating this quote: `A user interface is like a joke. If you have to explain it, it's not that good.` - Martin LeBlanc
> 
> Also another quote from Elon Musk:
> `Some people don't like change, but you need to embrace change if the alternative is disaster.`

3 YEARS AGO Joomla was in alpha stage and was tested only by developers. 
Now Joomla has become available to ordinary users who have encountered a problem. They do not understand what this or that parameter does. Previously, they could read the tooltip. There are no hints right now. What is understandable to the developer is not clear to the ordinary user who's blogging their cat's site

**Comment by dgrammatiko** [comment]
@Sulpher @PhocaCz @kernusr Let me educate you a little bit because by criticizing without the appropriate knowledge you're offending some people here, particularly Brian and me in this case.

The decision to declutter the UI was one of the main goals for Joomla 4.
The actionable tasks were (roughly speaking) two:
- Remove tooltips and inline the description where was needed (that was done)
- Introduce a better inplace help which I coarsely implemented as a possible way that this could be done back in 2017: https://github.com/joomla/40-backend-template/pull/247

The in-place help part then was given to a GSOC student in 2018, but the solution was rejected: https://github.com/joomla-projects/gsoc17_helpscreens_on_jdocs

The in-place help was given again to a GSOC student this year and I have no clue what's the status: https://github.com/joomla-projects/soc21_guided-tour

In short, the project had a very viable solution that would cover every requirement that was given. Unfortunately, you are criticizing something with a very poor background on the subject.
YES right now the help part is suffering but this is due to the lack of contributors.

Ideas are not exactly contributions, you want to help then present some code...

And I'm unsubscribing...

**Comment by PhocaCz** [comment]
> @Sulpher @PhocaCz @kernusr Let me educate you a little bit because by criticizing without the appropriate knowledge you're offending some people here, particularly Brian and me in this case.

@dgrammatiko Please let me know where exactly I offended you?

Can you distinguish between criticism and feedback? If we are giving some feedback without some knowledge, then surely there is some rational argument and certainly not an argument like "The change WAS DONE 3 YEARS AGO where was your feedback all these years?" Only now will I make the first criticism. Instead of addressing what is in the very first post, i.e. the missing parameter description, there is only misrepresentation and nonsensical arguments with personal attacks, but certainly not from our side.

So can we get back to the point? **The missing parameter description**.

**Comment by nikosdion** [comment]
You’re all wasting your time. It’s like talking to a stone wall. No matter what you say and when you’ll be told you’re wrong. That’s the one true Joomla way of dealing with its contributors. 

Three years ago Joomla was alpha 2 and every comment about the UI was met with “this is alpha, it is not the final version, we are still making changes to the template”. Now you’re told you should’ve provided feedback back when it was blatantly ignored. 

Two years ago it was late alphas when I contributed improvements to Download Keys. I added tooltips. I was asked to remove them because accessibility. Now you’re told they don’t have to do with accessibility per se. 

You’re told that the interface should be intuitive, yet Joomla has many non obvious options. We could phrase them with long sentences but this is against decluttering, the supposed main goal in J4. Meanwhile the documentation page for many core pages, including the Users options for example, is out of date and there is no indication what each option does. 

I think it’s pretty obvious what is going on. There has been ZERO UX work on Joomla 4, something which is blatantly obvious to those of us who were at the Joomla 4 sprint in Athens when the UX lead was told that the UX team is supposed to just say yes to everything the developers want to implement. That’s why she quit. She never told anybody but fuck it, she’s my wife now, she told me and I can finally say it after six years. That’s Joomla’s problem, right there. UX needs to drive development, not the other way around. UX is about what our target audience needs to get shit done, not any one person’s ego. I’ve implemented things in my software I personally disagree with EXACTLY BECAUSE the UX research said that my users actually needed those features to get stuff done. If people here can’t put their massive egos to the side to help the users that’s a systemic failure and the single most major endemic problem in Joomla.

**Comment by rdeutz** [comment]
I was also in Athens and can't remember that someone said the UX team should just say yes, anyway I don't know anything but this wasn't an offical statement from team. We all know we can do better in certain areas and I can understand the frustration on both sides. From what I know the guided tour project was good and successful. For the people missing information at some places, let's discuss and add them if they are providing something usefull. I can say that I am not a big fan of tooltips, if they have to be there, there then to often without information. I know this becuase have written to much of them :-)

**Comment by brianteeman** [comment]
>  For the people missing information at some places, let's discuss and add them if they are providing something useful

> Did we get it 100% right. Probably not but that is why we had beta versions and release candidates so that people could test and give feedback. If my memory serves me correctly it is three years since the change was made.

Nothing is preventing you having extra information for your fields. Just that core displays them and not hidden in an onhover tooltip.

Nothing is preventing anyone submitting a pull request (or creating an issue) to re-introduce a specific field description if something is not clear. There have already been a few of those over the last 3 years.

For the record the work to remove the descriptions (which I did on the specific request of the release lead) was not done automatically or unilaterally but carefully with every string reviewed before removal. IIRC it took about 70 hours of work and was deliberately done early in the process so that it could be reviewed. (Of course no one expected that process would take so long but the lack of regular contributors accounts for that)

**Comment by nikosdion** [comment]
Well, Robert, back then Marco was the J4WG lead — imposed by Tess, despite our objections, just because in her head she believed I was out to get her or something (never really understood why but apparently it was so deeply rooted that Matt Baylor stopped talking to me as well around the same time; go figure). Marco was the one who said to a room full of UX people that they should just follow what the developers say and the reason half the UX team including Crystal, its lead  and now my wife, resigned as soon as the sprint was over. What, do you think these UX professionals — some of them _very_ accomplished in the world of UX, who gave up their vacation time to fly around the world and help Joomla — up and left for no reason whatsoever? Joomla is now the shortest joke in the UX world and there's a guaranteed lack of interest from any professional UX researchers.

While this was happening you and all of us on the development side were arguing whether Joomla 4 should be a proper CMS (like all previous versions) or the bastard child of Laravel and Drupal. I will not say who was for and against to protect the innocent. I'm glad that eventually reason prevailed and we have a Joomla 4 CMS — for a while it was not a given and I was contemplating if I should go back to business consultancy when Joomla inevitably craters. Thank George (and Michael) for persevering.

Anyway, I am not surprised that this very critical piece of information did not flow down to the rest of the Joomla 4 Working Group. Joomla has been and still is a collection of insular fiefdoms. The only way I know what happened is that I married the former UX lead. I don't think it's reasonable for an organisation to require getting married to each other for the UX and development groups to have a viable communication channel.

Speaking of which, a few months ago Marco asked Crystal to once again lead the UX team — for Joomla 4.1. Naturally, Crystal asked if there were any guarantees that the product of the UX research would be used to make actual changes in Joomla. Marco said in no uncertain terms that there are **ABSOLUTELY NONE**. Crystal decided not to waste her time and ruin her reputation just so Joomla has the excuse that UX research was conducted (but never used, something nobody would ever admit) to justify stupid interface decisions and nonsensical features. Did that information flow to the rest of the leadership? I seriously doubt it and I'm not surprised it didn't for the reason stated above.

These project organisation and discipline problems are yours to solve as the president of OpenSourceMatters Inc. I do not envy your position. I also have to recuse myself from offering any advice because even though I would be able to do it impartially, wearing my business consultant hat on, this is most definitely not the way it would be perceived by those who have most to lose. If you can guarantee that this would not be a problem you can always hit me up for free consultancy; I am not trying to hypocritically run away from the problem, I was just being a realist about my status in the project and on the eyes of its stakeholders.

Regarding tooltips, I am not a big fan either. They have very bad accessibility — even when made accessible — as they interrupt the flow of the form. They have discoverability problems unless they are used everywhere consistently, something that is not even happening in Joomla 3, let alone 4.

I had the same problem with my own software, namely the restoration script in my backup software. I found a way — which can and SHOULD be improved upon — of adding help text below each field but having it hidden (from the screen AND screen readers) by default. I put a Show / Hide Help button at the top of the form. You're lost? Click the button, you get a quick reference below each field. Need more details? Hit the Help button in the toolbar to open the documentation. Form help text getting in your way? No problemo, hit the Show / Hide Help button again to hide it. It's definitely missing a keyboard shortcut for people with accessibility concerns but it's a workable middle ground solution.

As for adding to the translators workload: I am pretty sure our translators wouldn't mind translating a few hundreds more strings of meaningful text which would help hundreds of thousands of people make sense of the Joomla user interface. It's much better than having to translate three Block FLoC strings for a feature which does absolutely nothing and will most likely never do anything of any import since Google seems to not even be renewing let alone expanding their experimental deployment which was limited to Northern American users anyway.

**Comment by nikosdion** [comment]
BTW, there is one thing that was not addressed at all: contributors always being told they are wrong whether they do X or NOT-X. This is madness.

While there are some rampant masochists like yours truly who keep coming back after a good beating I don't think the world at large suffers from the same malady I do. Most good developers who were once contributing to Joomla have now stopped even bothering to try because they know the inevitable waste of time and energy that ensues.

As I have said since 2009 and put again in writing in 2015, the only way to address that is by having a solid roadmap and someone where the buck stops — call them an architect, a lead developer or Clint fucking Eastwood, I don't care about the label. While we have neither everyone and their dog will put their ego first and the one with the loudest voice and people who benefit from their agenda in the project will get their way. That's no way to run a FOSS project. At best, it's a thinly veiled excuse to hire the same people who berate contributors to supposedly work on the product. We tried that once and it took a massive volunteer drive and 6 months to get something halfway stable out the door after ~2 years of paid development. It took another 6 months of hard work by volunteers to make it halfway decent.

Listen to your users. They are not always right, but at least you can ask them what is their use case before saying “you don't know what you're talking about” or ”you should have spoken up three years ago when we were actively ignoring you”. I have been using Joomla since it was called Mambo and I still have several WTAF moment with the new parameters. For example, the “Only Use In Subforms” option in fields is not documented, doesn't have a tooltip and does nothing meaningful. I have a PR which makes it do something meaningful. Is it right? I have NO IDEA WHATSOEVER because the intent is not conveyed by code, there is no tooltip and no documentation. I had to go with my gut. Again, I am using this darned thing for just over 16 years now and _I am confused_. Can you imagine someone who just started using Joomla now? They look at the options and they're like “the hell is this, I'm running away”.

That's where I stand, in case anyone had any doubts whatsoever.

**Comment by Sulpher** [comment]
Well, this discussion is nonsense and the last drop for me.
And let me say my final thoughts.
I see the President of OSM has read this topic and did try to find a constructive solution. And Nickolas provided details concerning background relations and even added some funny stories to add a little positive way (thanks).
However, the problem is not resolvable until the community is not cleared from toxic persons. 
Let's look at this discussion from the beginning. 
A problem was identified and I suggested talking about advantages and disadvantages and suggested the solution in a friendly and polite maneer. English is a rich language and greatly allows expressing emotions, so words matter and how they can be said.  Please note that the idea suggested here was posted in POLITE MANEER.

And what did I get? Tons of negative just for daring to say a word!
Some persons started demagogy and attacked me personally here and blaming for not testing Joomla 4 three years ago, took the role of a teacher (not being a UX specialist) in his arrogant manner. And those persons are behind the project. 
Enough! I will let nobody break my day because of their extremely negative nature.

Obviously, the Joomla community has people with mental problems and we, the persons who do something for the project, are neither psychotherapists to listen to such delirium tremens nor flushing cisterns to neutralize negative emotions of persons who lost the sense of reality a long time ago. 

It is not a secret that Joomla is in trouble and has been losing users each year. The reasons are different but the problem is in TOXIC persons who have bitch behaviour and have been destroying the community spirit with personal attacks on everyone who disagrees.

Joomla 4 stable is out. Joomla needs help as never had. Every contribution matter. 
Is Joomla means 'we are together as a whole?'. NO!  In fact **we are near but not together**.  "Thanks" to toxic persons.
Be well and goodbye. I am leaving this discussion.

**Comment by rdeutz** [comment]
After this very long time of Joomla 4 development, in which a lot of people have spent a lot of time I hope that we all can come back to a productive and friendly development situation. Joomla4 has a lot of potential and we all can make it better. I think it is clear that a tooltip with not more information is useless. Replacing all was a good step but I also agree that we should look at this from a different angle and might add more explaining text when it makes sense. As someone who knows Joomla very good some things might be obvious and got removed but there is an easy way back. A PR can make a change.

So please, if you think that's missing open a PR or an issue it's always easier to talk about something concret than about something philosophical. Converting tooltips into viewalble text was a good move but it might went to far. Better we find it know than never, much better if we have found it years ago but that didn't happen. That's the reality.

I am locking this conversation, I think needed things are said, let's do something concret to make Joomla better for all of us.
